### DISCUSSION TITLE
Drop Video on Canvas

### DISCUSSION THREAD

**Discussion by dany-dev** [GrapesJS/grapesjs#4987] [Q&A] [answered]
Hello @artf, first I would like to thank you for this amazing framework.

So I was wondering that from other sites or application I can drag text and image and drop them in canvas and they will be added. But this doesn't work for video. 

Can you please some code or steps to enable a  feature to drop video just like image. 

Thank you.

**Comment by dany-dev** [comment, marked as answer]
Hello @artf, any solution. Thank you.

**Reply by artf** [reply]
Check this one https://github.com/GrapesJS/grapesjs/discussions/3849

**Reply by dany-dev** [reply]
Thanks a lot. It worked!

**Comment by dany-dev** [comment]
Hello @artf, now I am having another issue. when I use `canvas:dragdata` event, I want to run an ajax to upload the file to server, I have used Promise and async await function but didn't succeed. 
![image](https://user-images.githubusercontent.com/35777582/226443907-aa2cf0af-65a5-494b-a915-45c58c4bd4db.png)
This is the error I recieved and not wait for ajax request to complete, please help.

Thank you.

**Reply by artf** [reply]
Can you show your code used in `canvas:dragdata`?
